### PR TITLE
Feat(parser): add a file extention property

### PR DESCRIPTION
**Pull request by takker99**
## Proposed Changes

- add the property `lang` in `CodeBlock`
- enable to parse filenames like `hello.ts(js)`

### CODE REVIEW

File: src/block/CodeBlock.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ export interface CodeBlock {
   indent: number
   type: 'codeBlock'
   fileName: string
+  lang: string
```
Comment (progfay) [inline src/block/CodeBlock.ts:12]: Language of CodeBlock is not simple.
For example, language of `code:cljc` is clojure, `code:processing` is java in Scrapbox.

This change only provides the file extension, and property name `lang` is not suitable.

**Comment by progfay** [comment]
Thank you for your Pull Request!

But, this package will not support for file extensions.
For getting extension of filename in CodeBlock, try to parse filename yourself.
If you use this package in Node.js, [`path.extname`](https://nodejs.org/api/path.html#path_path_extname_path) is very useful.

File: src/block/CodeBlock.ts (hunk 1 above)
Comment (takker99) [reply src/block/CodeBlock.ts:12]: The behavior of `lang` is equal to `scrapbox.Page.lines`, so I thought it was natural to push a file extension into `lang`.
- For example:
- [Image from Gyazo](https://gyazo.com/200993ad37521f63b7172396eb8b26f9)
But I also understand property name lang is not suitable. I'm going to close this PR.

**Comment by takker99** [comment]
Thank you for your review!

At least I wanted you to implement parsing filenames like `script.gs(js)`  so that I recreated code block URLs easily.
But I understand the reason you will not support file extensions. I'll parse file extensions and `script.gs(js)` by myself.

**Pull request closed without merging**